src/ngraph/runtime/cpu/op/as_cast.hpp

```diff
@@ -0,0 +1,206 @@
+#include "ngraph/op/sum.hpp"
+#include "ngraph/op/tan.hpp"
+#include "ngraph/op/tanh.hpp"
+#include "ngraph/op/topk.hpp"
+
+#define BACKEND_OPS                                                                                \
+    NGRAPH_OP(BatchDot, ngraph::op)                                                                \
+    NGRAPH_OP(BatchNormInferenceRelu, ngraph::op)                                                  \
+    NGRAPH_OP(BatchNormTrainingRelu, ngraph::op)                                                   \
+    NGRAPH_OP(BoundedRelu, ngraph::op)                                                             \
+    NGRAPH_OP(ConvertLayout, ngraph::runtime::cpu::op)                                             \
+    NGRAPH_OP(ConvolutionAdd, ngraph::op)                                                          \
+    NGRAPH_OP(ConvolutionBiasAdd, ngraph::op)                                                      \
+    NGRAPH_OP(ConvolutionBiasBackpropFiltersBias, ngraph::op)                                      \
+    NGRAPH_OP(ConvolutionBias, ngraph::op)                                                         \
+    NGRAPH_OP(ConvolutionRelu, ngraph::op)                                                         \
+    NGRAPH_OP(GroupConvolutionBias, ngraph::op)                                                    \
+    NGRAPH_OP(GroupConvolution, ngraph::op)                                                        \
+    NGRAPH_OP(LeakyRelu, ngraph::op)                                                               \
+    NGRAPH_OP(LoopKernel, ngraph::runtime::cpu::op)                                                \
+    NGRAPH_OP(Lstm, ngraph::op)                                                                    \
+    NGRAPH_OP(MatmulBias, ngraph::op)                                                              \
+    NGRAPH_OP(MaxPoolWithIndicesBackprop, ngraph::op)                                              \
+    NGRAPH_OP(MaxPoolWithIndices, ngraph::op)                                                      \
+    NGRAPH_OP(QuantizedConvolutionBiasAdd, ngraph::op)                                             \
+    NGRAPH_OP(QuantizedConvolutionBias, ngraph::op)                                                \
+    NGRAPH_OP(QuantizedConvolutionBiasSignedAdd, ngraph::op)                                       \
+    NGRAPH_OP(QuantizedConvolutionRelu, ngraph::op)                                                \
+    NGRAPH_OP(Rnn, ngraph::op)                                                                     \
+    NGRAPH_OP(SigmoidMultiplyBackprop, ngraph::op)                                                 \
+    NGRAPH_OP(SigmoidMultiply, ngraph::op)                                                         \
+    NGRAPH_OP(UpdateSlice, ngraph::op)
+
+#define NGRAPH_OP(a, b) a,
+enum class OP_TYPEID
+{
+#include "ngraph/op/op_tbl.hpp"
+    BACKEND_OPS
+};
+#undef NGRAPH_OP
+
+template <OP_TYPEID tid>
+struct Hash
```

Comment:
`Hash` could be named better to reflect the fact that this class is used as template function, whose input is a number tid and output is a type. How about `MapTypeIDToOp`